crates/codegen/parser_runtime/src/support/precedence_helper.rs

```diff
@@ -0,0 +1,130 @@
+use super::{
+    super::{cst, kinds::RuleKind},
+    parser_result::ParserResult,
+};
+
+pub struct PrecedenceHelper;
```

Comment:
IIUC, this PR also fixes #562 and #563

Should we link them and add test cases?

Reply:
Link them?

Reply:
I mean linking the issue to the PR. I did that here as an example:

<img width="341" alt="image" src="https://github.com/NomicFoundation/slang/assets/15987992/3924d22c-2619-48eb-828c-3f49bec64d50">

I suggest adding new test cases to cover the fix, since our existing tests were succeeding.

Reply:
I updated the existing tests - the problem was we approved the tests that actually were erroenous parses.

Reply:
Thanks!